### PR TITLE
Updates for 12.10.0 and 12.10.1

### PR DESCRIPTION
**Pull request by Marie-Terrell**
Added the fix fix from SH-57

### CODE REVIEW

**Comment by olaurino** [comment]
Thanks! It looks you probably the username/email you configured locally for git do not match the ones in github, so the commits do not appeared as originated by your github account. Not a problem per se, but something to make sure next time you push something.

**Review by olaurino — APPROVED** [review]

**Comment by Marie-Terrell** [comment]
Oops. Thank you!

File: recipes/xspec/build.sh [hunk 1]
```diff
@@ -23,19 +23,22 @@ if [ -n "$XSPEC_PATCH" ]
...
     cd ${XSPEC_MODELS_ONLY}/Xspec/src;
     curl -LO -z ${XSPEC_PATCH} http://heasarc.gsfc.nasa.gov/docs/xanadu/xspec/issues/archive/${XSPEC_PATCH};
-    curl -LO -z ${XSPEC_PATCH_INSTALLER} http://heasarc.gsfc.nasa.gov/docs/xanadu/xspec/issues/archive/${XSPEC_PATCH_INSTALLER};
+    curl -LO -z ${XSPEC_PATCH_INSTALLER} http://heasarc.gsfc.nasa.gov/docs/xanadu/xspec/issues/${XSPEC_PATCH_INSTALLER};
     tclsh ${XSPEC_PATCH_INSTALLER} -m -n;
     rm -rf XSFits;
     cd ${XSPEC_DIR};
 fi
 
+#Copy in the OGIPTable fix
```
Comment (DougBurke) [inline recipes/xspec/build.sh:32]: I haven't been following this too closely, but I believe the OGIPTable fix is *only* for 12.10.0 and should not be applied to 12.10.1. Am I right in thinking it's currently applied to 12.10.1?

Comment (Marie-Terrell) [reply recipes/xspec/build.sh:32]: I did keep the the OGIPTable fix for 12.10.1. If this is no longer needed, I can remove it and re-build.

Comment (olaurino) [reply recipes/xspec/build.sh:32]: Before rebuilding I would check the differences in the files. If there are no significant differences there is no need to rebuild.

**Comment by DougBurke** [comment]
Long term I'm just concerned that this is a "patch" that could keep being
installed once not needed and may eventually cause surprising behavior. If
there's no change then for the short term it's okay

**Comment by olaurino** [comment]
Just to clarify, I agree with that. Just saying that given the usage of these binaries I wouldn't find it necessary to rebuild them unless really required. But we should indeed remove the patching if it doesn't apply anymore.

**Comment by Marie-Terrell** [comment]
So, here are the two files (the 12.10.0 fixed file and the 12.10.1 file). There are differences between the two files. It looks like a change from std::auto_ptr to std::unique_ptr and replacing std::ostringstream with tcout. I thought tcout was the cause of the issues in the past (SH-57).
[OGIPTable_12.10.1.txt](https://github.com/sherpa/xspec-build/files/2837335/OGIPTable_12.10.1.txt)
[OGIPTable_12.10.0_fix.txt](https://github.com/sherpa/xspec-build/files/2837336/OGIPTable_12.10.0_fix.txt)

**Comment by DougBurke** [comment]
XSPEC 12.10.1 introduces a new function for handling table models - this is
used in my PR [1] - which I believe "works around" the tcout issue we saw
in 12.10.0.

[1]
https://github.com/sherpa/sherpa/pull/534/commits/7f643c920cce3c2a88b2189af792d781cac942e1

**Pull request merged**